tests/ConnectionTest.php

```diff
@@ -172,6 +173,35 @@ public function testCommitStartsTransactionInNoAutoCommitMode(): void
         self::assertTrue($conn->isTransactionActive());
     }
 
+    public function testBeginTransactionFailureAfterCommitInNoAutoCommitMode(): void
+    {
+        $driverConnectionMock = $this->createMock(DriverConnection::class);
+        $driverConnectionMock->expects(self::exactly(2))
+            ->method('beginTransaction')
+            ->willReturnOnConsecutiveCalls(
+                true,
+                self::throwException(new RuntimeException()),
+            );
+
+        $driver = self::createStub(Driver::class);
+        $driver
+            ->method('connect')
+            ->willReturn(
+                $driverConnectionMock,
+            );
+        $conn = new Connection([], $driver);
+
+        $conn->setAutoCommit(false);
+
+        $conn->getServerVersion(); // causes a call to connect(), which is protected
```

Comment:
@simPod do you have a better idea for this?

Reply:
Since we can't use beginTransaction, this will do. Or `getNativeConnection` but there's no difference.

```suggestion
        $conn->getServerVersion(); // Invoke protected `connect()`
```

Reply:
```suggestion
        /** Invoke protected {@see Connection::connect()} */
        $conn->getServerVersion();
```